### ISSUE TITLE
error.response undefined in simple interceptor

### ISSUE THREAD

**Issue by SeeThruHead** [axios/axios#1434]
Axios v0.17.1
```
instance.interceptors.response.use(res => res, error => {
  console.log('ERROR');
  console.log(error);
  console.log('response:');
  console.log(error.response);
});
```
result
```
ERROR
index.js:47 Error: Network Error
    at createError (createError.js:16)
    at XMLHttpRequest.handleError (xhr.js:87)
index.js:48 response:
index.js:49 undefined
```

confused at to why this is happening, i'm trying to retry requests based on error.response.status but error.response isn't always defined. Is this expected behaviour? If so what is your recommended fix?

**Comment by rod-stuchi** [comment]
The same exact issue here

**Comment by rod-stuchi** [comment]
@SeeThruHead do you have any .env files setting out `timeout` or `baseurl` ?

**Comment by johncloud200** [comment]
I also facing this, did you find solution?

**Comment by daniel-lawrence** [comment]
@SeeThruHead I had this issue when the response did not include the `Access-Control-Allow-Origin` header. The error response was received, but disallowed by CORS. When that happens, it looks like axios doesn't actually set `error.response`, even though there was a received response.

So, at least in my case, the fix was to make sure `Access-Control-Allow-Origin: *` was included in the response's headers when there was an error.

**Comment by AllanPinheiroDeLima** [comment]
I'm running this issue too. 
I want to control user logged in state into interceptors. For instance, if my interceptors catch a response with status 401, I want to logoff / redirect this user.

**Comment by thehaytonproject** [comment]
Stumbled across this thread whilst debugging the exact same issue.

Turns out my problem was of my own doing. I had a misconfigured **request** interceptor where I neglected to return the config. I _think_ this was throwing an error that was caught by the response interceptor, even though a request had not been made. I'm pretty sure this can be triggered by any code chained to a request, for example chained promises.

Not sure if this is intentional or not, but something worth checking.
